modules/server/src/main/ts/bedrock/auto/DriverLoader.ts

```diff
@@ -102,78 +62,29 @@ export const loadDriver = (browserName: string, settings: DriverSettings): Drive
   }
 
   const execNames = browserExecutables[browserName] || [];
-  const execPath = findExecutable(execNames);
+  const execPath = ExecUtils.findExecutable(execNames);
   if (execPath !== null) {
-    return execLoader(execPath);
+    return ExecUtils.execLoader(execPath);
   } else {
-    throw new Error('Unable to find a suitable driver for ' + browserName);
+    throw new Error('Local1: Unable to find a suitable driver for ' + browserName);
   }
 };
 
-export const waitForAlive = (proc: ChildProcess | null, port: number, timeout: number, path: string): Promise<void> => {
-  const url = 'http://127.0.0.1:' + port + path + '/status';
-  console.log('waiting for alive @: ', url);
-  const start = Date.now();
-  return new Promise<void>((resolve, reject) => {
-    let timeoutId: ReturnType<typeof setTimeout> | null = null;
-
-    const onStartError = (error: string) => {
-      if (timeoutId) {
-        clearTimeout(timeoutId);
-      }
-      reject('Driver failed to start (' + error + ')');
-    };
-
-    const onServerError = (err: string) => {
-      if (Date.now() - start > timeout) {
-        reject('Timed out waiting for the webdriver server. Error: ' + err);
-      } else {
-        // eslint-disable-next-line @typescript-eslint/no-use-before-define
-        timeoutId = setTimeout(checkServerStatus, 50);
-      }
-    };
-
-    const checkServerStatus = () => {
-      http.get(url, (res) => {
-        if (res.statusCode === 200) {
-          let rawData = '';
-          res.on('data', (chunk) => rawData += chunk);
-          res.on('end', () => {
-            try {
-              const data = JSON.parse(rawData);
-              if (data.value.ready || data.status === 0) {
-                if (proc) {
-                  proc.removeListener('exit', onStartError);
-                  proc.removeListener('error', onStartError);
-                }
-                resolve();
-              } else {
-                onServerError('Not ready to accept connections');
-              }
-            } catch (e) {
-              onServerError(e.message);
-            }
-          });
-        } else {
-          onServerError('Received non 200 status (' + res.statusCode + ')');
-        }
-      }).on('error', onServerError);
-    };
-
-    // Bind process listeners
-    if (proc) {
-      proc.on('exit', onStartError);
-      proc.on('error', onStartError);
-    }
+export const startAndWaitForAlive = async (driverSpec: DriverSpec, port: number, timeout = 30000): Promise<void> => {
+  const api = driverSpec.driverApi;
+  
+  // Start the driver
+  let args;
+  if (api.isNpm) {
+    args = {port};
+  } else {
+    args = ['--port=' + port];
+  }
```

Comment:
```suggestion
  const args = api.isNpm ? { port } : [ '--port=' + port ];
```
If this is not to be changed. Might still need typing.

Reply:
Types for that `driverApi.start` have not been an easy task due to it receiving rest parameters and typescript being weird about it. But at least it got inlined as suggested